todo_app/app.py

```diff
@@ -1,55 +1,118 @@
-from operator import itemgetter
+from operator import attrgetter
 from flask import Flask, render_template, request, redirect
-from todo_app.data import session_items
 from todo_app.flask_config import Config
+import os,requests
+from dotenv import load_dotenv
+from todo_app.ToDoItem  import  ToDoItem
 
+# Set app variables
+load_dotenv()
+key=os.getenv('TRELLO_KEY')
+token=os.getenv('TRELLO_TOKEN')
+trelloBoardID='60fad7c6c463fc540fbffb94'
+trelloBaseList='60fad7c6c463fc540fbffb95' # will place new cards in To Do list
 app = Flask(__name__)
 app.config.from_object(Config)
 
+# Dictionary used to map request calls 
+jsonDict={
+    "PUT":requests.put,
+    "POST":requests.post,
+    "GET":requests.get,
+    "DELETE":requests.delete
+ }
 
+# Common function used to request data from URL's
+def requestJson (method, url, args):
+    r = jsonDict[method](url, params=args) 
+    r.raise_for_status()
+    #Should handle error statuses - TODO
+    return r.json()
+
+#Main functions/app urls
+
+# Main Page - will query for cards and display
 @app.route('/')
 def index():
-    items=(sorted(session_items.get_items(), key=itemgetter('status'),reverse=True))
-    return render_template('index.html',list=items)
-
+    cards=getTrelloCards(trelloBoardID, key, token, False)
+    return render_template('index.html',list=cards)
 
+# Used to add new item
 @app.route('/add', methods=['POST'])
 def add_item():
-    
-    user_item=request.form.get('item_name')
-    if not user_item == "":
-        session_items.add_item(user_item)
+    card_title=request.form.get('item_name')
+    card_desc=request.form.get('item_desc')
+    card_due=request.form.get('item_due')
+    if not card_title == "": 
+        cardDetails={"idList":trelloBaseList,"name":card_title,"pos":"top","desc":card_desc,"due":card_due}
+        updateTrelloCard('', cardDetails, "POST", key, token) 
+        getTrelloCards(trelloBoardID, key, token, True)   
     return redirect("/")
 
+# Used to sort cards being displayed
+@app.route('/sort', methods=['POST'])
+def sort_item():
+    attr=str(request.form.get('sortBy'))
+    if attr == "reset":
+        getTrelloCards(trelloBoardID, key, token, True)
+        return redirect("/")
+    ToDoItem.cardsList=sorted(ToDoItem.cardsList, key=attrgetter(attr))
```

Comment:
Rather than persisting a list of cards on `ToDoItem.cardsList`, you might want to fetch/sort the cards each time.
You could store sorting preference in a cookie (using Flask's `session`), rather than updating this backing list. 

That way:
- The server could serve multiple users without them interfering with each others requests
- Using this endpoint doesn't implicitly depend on having an up to date cardsList